okta/resource_user.go

```diff
@@ -0,0 +1,672 @@
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User honorific suffix",
+			},
+			"last_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "User last name",
+			},
+			"locale": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User default location",
+			},
+			"login": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "User Okta login (must be an email address)",
+				ForceNew:     true,
+				ValidateFunc: matchEmailRegexp,
+			},
+			"manager": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Manager of User",
+			},
+			"manager_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Manager ID of User",
+			},
+			"middle_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User middle name",
+			},
+			"mobile_phone": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User mobile phone number",
+			},
+			"nick_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User nickname",
+			},
+			"organization": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User organization",
+			},
+			"postal_address": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User mailing address",
+			},
+			"preferred_language": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User preferred language",
+			},
+			"primary_phone": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User primary phone number",
+			},
+			"profile_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User online profile (web page)",
+			},
+			"second_email": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User secondary email address, used for account recovery",
+			},
+			"state": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "User state or region",
+			},
+			"status": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The status of the User in Okta - remove to set user back to active/provisioned",
+				Default:      "ACTIVE",
+				ValidateFunc: validation.StringInSlice([]string{"ACTIVE", "STAGED", "DEPROVISIONED", "SUSPENDED"}, false),
+				// ignore diff changing to ACTIVE if state is set to PROVISIONED
+				// since this is a similar status in Okta terms
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if old == "PROVISIONED" && new == "ACTIVE" {
```

Comment:
This is totally not necessary to change, just noticed this could be a one liner `return old == "PROVISIONED" && new == "ACTIVE"` :-D

Reply:
totally right, this is redundant; fixing this up!